keras/engine/saving.py

```diff
@@ -244,73 +199,223 @@ def convert_custom_objects(obj):
             return custom_objects[obj]
         return obj
 
-    opened_new_file = not isinstance(filepath, h5py.Group)
-    if opened_new_file:
-        f = h5py.File(filepath, mode='r')
+    model_config = f['model_config']
+    if model_config is None:
+        raise ValueError('No model found in config.')
+    model_config = json.loads(model_config.decode('utf-8'))
+    model = model_from_config(model_config, custom_objects=custom_objects)
+    model_weights_group = f['model_weights']
+
+    if 'keras_version' in model_weights_group:
+        original_keras_version = model_weights_group['keras_version'].decode('utf8')
+    else:
+        original_keras_version = '1'
+    if 'backend' in model_weights_group:
+        original_backend = model_weights_group['backend'].decode('utf8')
+    else:
+        original_backend = None
+
+    layer_names = model_weights_group['layer_names']
+
+    layers = model.layers
+
+    filtered_layers = []
+    for layer in layers:
+        weights = layer.weights
+        if weights:
+            filtered_layers.append(layer)
+
+    filtered_layer_names = []
+    for name in layer_names:
+        layer_weights = model_weights_group[name]
+        weight_names = layer_weights['weight_names']
+        if weight_names:
+            filtered_layer_names.append(name)
+
+    layer_names = filtered_layer_names
+    if len(layer_names) != len(filtered_layers):
+        raise ValueError('You are trying to load a weight file '
+                         'containing ' + str(len(layer_names)) +
+                         ' layers into a model with ' +
+                         str(len(filtered_layers)) + ' layers.')
```

Comment:
Suggest to use `%` style
```python
        raise ValueError('You are trying to load a weight file '
                         'containing %d layers into a model with '
                         '%d layers.' % (len(layer_names), len(filtered_layers)))
```

Reply:
We don't have a unified way, but I think str.format is more common?